openpype/hosts/houdini/hooks/set_default_display_and_view.py

```diff
@@ -0,0 +1,67 @@
+from openpype.lib.applications import PreLaunchHook, LaunchTypes
+
+
+class SetDefaultDisplayView(PreLaunchHook):
+    """Set default view and default display for houdini via OpenColorIO.
+
+    Houdini's defaultDisplay and defaultView are set by
+    setting 'OCIO_ACTIVE_DISPLAYS' and 'OCIO_ACTIVE_VIEWS'
+    environment variables respectively.
+
+    More info: https://www.sidefx.com/docs/houdini/io/ocio.html#set-up
+    """
+
+    app_groups = {"houdini"}
+    launch_types = {LaunchTypes.local}
+
+    def execute(self):
+
+        OCIO = self.launch_context.env.get("OCIO")
+
+        # This is a cheap way to skip this hook if either global color
+        # management or houdini color management was disabled because the
+        # OCIO var would be set by the global OCIOEnvHook
+        if not OCIO:
+            return
+
+        houdini_color_settings = \
+            self.data["project_settings"]["houdini"]["imageio"]["workfile"]
+
+        if not houdini_color_settings["enabled"]:
+            self.log.info(
+                "Houdini workfile color management is disabled."
+            )
+            return
+
+        # 'OCIO_ACTIVE_DISPLAYS', 'OCIO_ACTIVE_VIEWS' are checked
+        # as Admins can add them in Ayon env vars or Ayon tools.
+
```

Comment:
This could be re factored into a method:
```py

default_display = houdini_color_settings["default_display"]
if default_display:
	self._set_context_env("OCIO_ACTIVE_DISPLAYS", default_display)


default_view = houdini_color_settings["default_view"]
if default_view:
	self._set_context_env("OCIO_ACTIVE_VIEWS", default_view)


def _set_context_env(self, env_var, values_to_add):
	original_value = self.launch_context.env.get(env_var.uppercase(), "")
	new_value = ":".join(val for val in [values_to_add, env_var])
	self.log.info("Setting {} environment to: {}".format(env_var.uppercase(), new_value))
	self.launch_context.env[env_var.uppercase()] = new_value
```